strawberry/fastapi/router.py

```diff
@@ -30,10 +31,30 @@ async def __get_root_value():
         return None
 
     @staticmethod
-    async def __get_context(
-        background_tasks: BackgroundTasks, request: Request = None, ws: WebSocket = None
-    ) -> Optional[Any]:
-        return {"request": request or ws, "background_tasks": background_tasks}
+    def __get_context_getter(
+        custom_getter: Callable[..., Optional[Dict[str, Any]]]
+    ) -> Callable[..., Dict[str, Any]]:
+        async def dependency(
+            background_tasks: BackgroundTasks,
+            request: Request = None,
+            ws: WebSocket = None,
+            custom_getter: Dict[str, Any] = {},
+        ):
+            return {
+                "request": request or ws,
+                "background_tasks": background_tasks,
+                **custom_getter,
+            }
+
+        sig = signature(dependency)
+        sig = sig.replace(
+            parameters=[
+                *list(sig.parameters.values())[:-1],
+                sig.parameters["custom_getter"].replace(default=Depends(custom_getter)),
+            ]
+        )
```

Comment:
There's an alternative solution to basically build all the function. Something between those:

- https://github.com/Kludex/fastapi-authorization/blob/67765b28bd350bfef83288c4e4c36c55725f255a/fastapi_authorization/abac.py#L32-L45
- https://github.com/Kludex/fastapi-health/blob/f012f5a1e7a2e184fd7a51a17cf4c26215bb605e/fastapi_health/route.py#L103-L122 

But the current one is fine. :+1: